### DISCUSSION TITLE
bfe_module目录下有bfe_module.go和bfe_plugin.go基础创建区别

### DISCUSSION THREAD

**Discussion by gengchao** [bfenetworks/bfe#902] [Q&A (问题和回答)] [answered]
bfe_module目录下有bfe_module.go和bfe_plugin.go 两种加载模块的方式，他们有啥区别？什么情况下使用bfe_module.go加载模块，什么情况下使用bfe_plugin.go加载模块呢？

**Comment by mileszhang2016** [comment, marked as answer]
1. bfe_module是BFE最主流的扩展模块机制。所写的扩展模块代码需要和BFE其它代码一起编译，最后得到一个可执行文件。
    详情见 https://github.com/baidu/bfe-book/blob/version1/design/module/module.md

2. bfe_plugin用于建立“可动态加载的扩展模块“。所写的扩展模块代码可以独立编译为一个.so文件，然后和bfe的主程序一起运行。
   bfe_plugin是bfe_module的一个扩展，其回调机制和bfe_module是相同的。
   可以看 https://github.com/bfenetworks/bfe-dynamic-plugin 中的例子。

**Reply by gengchao** [reply]
明白了，谢谢~
